#! /bin/sh /usr/share/dpatch/dpatch-run
## escape_Lucene.Net_search_string_r148946.dpatch by Jo Shields <directhex@apebox.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Backport of upstream SVN revision r148946. Fixes a crash in the documentation
## DP: browser when searching for things like "()"

@DPATCH@
--- branches/mono-2-4/mcs/tools/monodoc/Lucene.Net/Lucene.Net/QueryParser/QueryParser.cs	2009/01/20 16:20:03	123891
+++ branches/mono-2-4/mcs/tools/monodoc/Lucene.Net/Lucene.Net/QueryParser/QueryParser.cs	2009/12/30 20:05:46	148946
@@ -108,7 +108,7 @@
 		static public Query Parse(System.String query, System.String field, Analyzer analyzer)
 		{
 			QueryParser parser = new QueryParser(field, analyzer);
-			return parser.Parse(query);
+			return parser.Parse(Escape(query));
 		}
 		
 		/// <summary>Constructs a query parser.</summary>
